net/dog/Portfile

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        ogham dog 0.1.0 v
+revision            0
+
+homepage            https://dns.lookup.dog
+
+description         ${name} is a command-line DNS client
+
+long_description    Dogs can look up! dog is a command-line DNS client, \
+                    like dig. It has colourful output, understands normal \
+                    command-line argument syntax, supports the DNS-over-TLS \
+                    and DNS-over-HTTPS protocols, and can emit JSON.
+
+categories          net sysutils
+platforms           darwin
+license             MIT
+
+checksums           rmd160  712de148301083163d46a9849316c5adf99c8070 \
+                    sha256  97b749124e3bdcb1c3a97076a2fb3dee29dfc16a29c9f638bdae1060ad4e3d89 \
+                    size    271723
+
+maintainers         {gmail.com:herby.gillot @herbygillot} \
+                    openmaintainer
+
+github.tarball_from releases
+distname            ${name}-v${version}-x86_64-apple-darwin
```

Comment:
I see you are installing a binary, so you should set supported_archs to match.

Can it not be compiled from source?

Reply:
The `cargo` portgroup downloads the Rust deps as crate compressed files from `crates.io`.  One of the dependencies is a crate in source code form from a Github repo.  I had issues getting around that, but after taking another look, I see that the `cargo_fetch` portgroup supports `cargo.crates_github`, so I'll give it another go later on.

Reply:
@ryandesign here's what I was running into when trying to build from source: https://paste.z0k.xyz/f9c7409b7a1f

```
DEBUG: system:  cd "/opt/local/var/macports/build/_Users_runner_work_1_s_net_dog/dog/work/dog-0.1.0" && /opt/local/bin/cargo build --release --frozen -v -j4 
error: failed to get `byteorder` as a dependency of package `dns v0.1.0 (/opt/local/var/macports/build/_Users_runner_work_1_s_net_dog/dog/work/ogham-dog-445ed98/dns)`

Caused by:
  failed to load source for dependency `byteorder`

Caused by:
  Unable to update registry `https://github.com/rust-lang/crates.io-index`

Caused by:
  failed to update replaced source registry `https://github.com/rust-lang/crates.io-index`

Caused by:
  found a virtual manifest at `/opt/local/var/macports/build/_Users_runner_work_1_s_net_dog/dog/work/.home/.cargo/macports/mutagen-c7abc956a10e8a3e2cc71f21279ea0a42f7b7c10/Cargo.toml` instead of a package manifest
Command failed:  cd "/opt/local/var/macports/build/_Users_runner_work_1_s_net_dog/dog/work/dog-0.1.0" && /opt/local/bin/cargo build --release --frozen -v -j4 
Exit code: 101
```

`cargo` complans that it can't find the `byteorder` dependency even though both versions requested by both the main project and the `./dns` sub-directory's `Cargo.toml` files have been vendored in by MacPorts.  This is why I had simply decided to just use the pre-built binary.

The project builds fine when built outside of MacPorts.